model/role.go

```diff
@@ -162,8 +197,44 @@ func init() {
 		PERMISSION_SYSCONSOLE_WRITE_USERMANAGEMENT_TEAMS.Id,
 		PERMISSION_SYSCONSOLE_WRITE_USERMANAGEMENT_CHANNELS.Id,
 		PERMISSION_SYSCONSOLE_WRITE_USERMANAGEMENT_PERMISSIONS.Id,
-		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT.Id,
-		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_WEB_SERVER.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_DATABASE.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_ELASTICSEARCH.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_FILE_STORAGE.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_IMAGE_PROXY.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_SMTP.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_PUSH_NOTIFICATION_SERVER.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_HIGH_AVAILABILITY.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_RATE_LIMITING.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_LOGGING.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_SESSION_LENGTHS.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_PERFORMANCE_MONITORING.Id,
+		PERMISSION_SYSCONSOLE_READ_ENVIRONMENT_DEVELOPER.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_WEB_SERVER.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_DATABASE.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_ELASTICSEARCH.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_FILE_STORAGE.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_IMAGE_PROXY.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_SMTP.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_PUSH_NOTIFICATION_SERVER.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_HIGH_AVAILABILITY.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_RATE_LIMITING.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_LOGGING.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_SESSION_LENGTHS.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_PERFORMANCE_MONITORING.Id,
+		PERMISSION_SYSCONSOLE_WRITE_ENVIRONMENT_DEVELOPER.Id,
+		PERMISSION_READ_ELASTICSEARCH_POST_INDEXING_JOB.Id,
+		PERMISSION_READ_ELASTICSEARCH_POST_AGGREGATION_JOB.Id,
+		PERMISSION_TEST_SITE_URL.Id,
+		PERMISSION_RELOAD_CONFIG.Id,
+		PERMISSION_INVALIDATE_CACHES.Id,
+		PERMISSION_RECYCLE_DATABASE_CONNECTIONS.Id,
+		PERMISSION_TEST_ELASTICSEARCH.Id,
+		PERMISSION_CREATE_ELASTICSEARCH_POST_INDEXING_JOB.Id,
+		PERMISSION_CREATE_ELASTICSEARCH_POST_AGGREGATION_JOB.Id,
+		PERMISSION_PURGE_ELASTICSEARCH_INDEXES.Id,
+		PERMISSION_TEST_S3.Id,
+		PERMISSION_TEST_EMAIL.Id,
```

Comment:
You can remove these.
```suggestion
```
Line 171 in this file already gives the ancillary permissions.